scipy/interpolate/_ppoly.pyx

```diff
@@ -40,7 +40,7 @@ DEF MAX_DIMS = 64
 @cython.cdivision(True)
 def evaluate(double_or_complex[:,:,::1] c,
              double[::1] x,
```

Comment:
While we're at it, might also allow for `const double x`

Reply:
Done! Tests pass - Ill take on `PPoly.integrate` and `BPoly.from_derivatives` in another PR?

Reply:
Im not sure why _differential_evolution is failing - it doesnt seem to have an obvious connection to `PPoly` and doesnt fail locally